modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java

```diff
@@ -149,6 +149,14 @@ protected void doProcessAction(
 		String redirectURL = ParamUtil.getString(
 			actionRequest, "redirect", ddmFormInstanceSettings.redirectURL());
 
+		DDMFormSuccessPageSettings ddmFormSuccessPageSettings =
+			ddmForm.getDDMFormSuccessPageSettings();
+
+		if (Validator.isNotNull(redirectURL) ||
```

Comment:
You don't need to validate the same thing twice. Just include `hideDefaultSuccessMessage(actionRequest);` inside the `if` body and keep the `else`.

Reply:
That's not quite what I was thinking. What happens if there's no redirectURL? Would the success page never be verified?

Reply:
During tests, I didn't see the object _redirectURL_ being empty in different cases I tested. So, I assumed that the success page will always be verified. Both cases reported [here](https://github.com/liferay-forms/liferay-portal/pull/1961#discussion_r982860925) are covered.